### PR TITLE
chore: updating to solidity 0.8

### PR DESCRIPTION
**Pull request by pandadefi**
## Description

On our last call we discussed pushing solidity version to 0.8.10

## Checklist

- [x] I have run vyper and solidity linting
- [x] I have run the tests on my machine
- [x] I have followed commitlint guidelines
- [x] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

**Comment by pandadefi** [comment]
bumping to 0.8.11

File: README.md [hunk 1]
```diff
@@ -103,7 +103,7 @@ Any command `in code blocks` is meant to be executed from a Mac/Linux terminal o
       - Alternatively, go to File -> Preferences -> Settings
       - If you’re using WSL, go to the Remote [WSL] tab
       - Otherwise choose the Workspace tab
-        - Search for _Solidity_ and copy and paste _v0.6.12+commit.27d51765_ into the _Solidity: Compile Using Remote Version_ textbox
+        - Search for _Solidity_ and copy and paste _v0.8.11 into the _Solidity: Compile Using Remote Version_ textbox
```
Comment (fubuloubu) [inline README.md:106]: commit number?

File: brownie-config.yaml [hunk 2]
```diff
@@ -6,19 +6,18 @@ autofetch_sources: true
 
 # require OpenZepplin Contracts
 dependencies:
-  - OpenZeppelin/openzeppelin-contracts@3.1.0
+  - OpenZeppelin/openzeppelin-contracts@4.4.1
```
Comment (fubuloubu) [inline brownie-config.yaml:9]: what's the release history look like here? any noteworthy changes that impact us?

File: tests/functional/vault/test_withdrawal.py [hunk 3]
```diff
@@ -3,439 +3,439 @@
 MAX_UINT256 = 2 ** 256 - 1
 
 
-def test_multiple_withdrawals(token, gov, Vault, TestStrategy, common_health_check):
```
Comment (fubuloubu) [inline tests/functional/vault/test_withdrawal.py:6]: what's with all the commented out tests?

**Review by fubuloubu — COMMENTED** [review]
Also can you post a summary of gas changes and contract size utilization?

File: brownie-config.yaml (hunk 2 above)
Comment (pandadefi) [reply brownie-config.yaml:9]: I have tried to get an idea of the changes logs for every versions https://github.com/OpenZeppelin/openzeppelin-contracts/releases
 
v3.2.0 -> add proxy to the project
 v3.3.0 -> solidity 0.7
 v3.4.0 -> fix ERC777 potential reentrancy
v4.0.0  -> Solidity 0.8
 v4.1.0 -> UUPS proxies as a cheaper alternative to Transparent proxies
 v4.2.0 -> ERC20Votes
 v4.3.0 -> More governance stuff
 v4.4.0 -> Many small changes

**Comment by pandadefi** [comment]
|  | solidity 0.6 | solidity 0.8 |
| --- | --- | --- |
| TestStrategy `deploy` | 2433139 |  2299644|
| TestStrategy `clone` | 273952 |  271762|
| TestStrategy `harvest` | 166184 |  163633|

File: brownie-config.yaml (hunk 2 above)
Comment (fubuloubu) [reply brownie-config.yaml:9]: Just make sure you've explored all of our dependencies and dependency tree to ensure we are getting rugged on a tiny change that doesn't make sense or affects our current utilization.

Comment (pandadefi) [reply brownie-config.yaml:9]: I don't see any solidity dependencies on https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/package.json that shouldn't be an issue.

**Review by storming0x — COMMENTED** [review]
I think we should move ahead w this PR to target a tag called 0.4.3-sol8

this is important to update brownie strategy mix to sol 0.8.12 and support strategies to use latest version of the sol language

**Pull request closed without merging**